packages/interface/src/store/actions.js

```diff
@@ -46,14 +48,16 @@ export const enableComplementaryArea =
 		scope = normalizeComplementaryAreaScope( scope );
 		area = normalizeComplementaryAreaName( scope, area );
 
-		const isComplementaryAreaVisible = registry
-			.select( preferencesStore )
-			.get( scope, 'isComplementaryAreaVisible' );
+		if ( persist ) {
```

Comment:
I think a problem is the preferences store won't be set to the correct value for `isComplementaryAreaVisible` when `persist` is `false`. Users who visit for the first time continue to have `undefined` as the `isComplementaryAreaVisible` preference, so the `isActiveByDefault` isn't being respected. The `getActiveComplementaryArea` selector seems to return the preference store value in an early return, so I think it won't be returning the intended value.

IIRC, the main use case for `isActiveByDefault` is for those first time visitors.

Revisiting this, I'm wondering why a default preference value for `isComplementaryAreaVisible` isn't being set that corresponds to `isActiveByDefault`:
https://github.com/WordPress/gutenberg/blob/fa636dc125301270d7ea28c1b90ac00ef612be75/packages/edit-site/src/index.js#L64-L77

That seems like it'd work well for the first time visitors and it doesn't trigger any API requests. The name of the default area still has to be set in the interface store.

Another curiosity is the way the interface package has a `setDefaultComplementaryArea` action, but it seems unused from what I can tell. Probably for back compat. Makes me wonder if it can be used for this. 🤷